### PR TITLE
Adding spring security

### PR DESCRIPTION
**Pull request by LiveNathan**
All the tests pass!

### CODE REVIEW

File: src/main/java/cn/RecipeAPI/Models/CustomUserDetails.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    public String username;
+
+    @Column(nullable = false)
+    //make sure that the password is never sent out but can be read when creating a new user
```
Comment (NikitaBaranov) [inline src/main/java/cn/RecipeAPI/Models/CustomUserDetails.java:28]: Security first!

File: src/main/java/cn/RecipeAPI/Security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+        return optionalUser;
+    }
+
+    @Transactional(readOnly = true)
+    public CustomUserDetails getUserByUserId(Long userId) throws EntityNotFoundException {
+        CustomUserDetails user = userRepo.getReferenceById(userId);
+
+        //call unproxy() to ensure all related entities are loaded—no lazy load exceptions.
+        return (CustomUserDetails) Hibernate.unproxy(user);
```
Comment (NikitaBaranov) [inline src/main/java/cn/RecipeAPI/Security/CustomUserDetailsService.java:43]: Cool!

**Review by NikitaBaranov — APPROVED** [review]
Looks good 👍

File: src/main/java/cn/RecipeAPI/Security/CustomUserDetailsService.java (hunk 2 above)
Comment (LiveNathan) [reply src/main/java/cn/RecipeAPI/Security/CustomUserDetailsService.java:43]: Ha! Yeah, that's Ryan's code. I don't actually understand what's happening on line 43.

**Pull request merged**